pkg/autodiscovery/common/utils/annotations.go

```diff
@@ -72,8 +74,8 @@ func extractCheckTemplatesFromMap(key string, input map[string]string, prefix st
 	if err != nil {
 		return []integration.Config{}, fmt.Errorf("in %s: %s", instancePath, err)
 	}
-
-	return BuildTemplates(key, checkNames, initConfigs, instances), nil
+	ignoreAdTags, _ := strconv.ParseBool(input[prefix+ignoreAutodiscoveryTags])
```

Comment:
does it return false if the parsing failed? 
maybe you can add a comment saying that if the parsing fail, it will return false.

Reply:
I added a comment in the last commit